examples/threathunter/threat_intelligence/ThreatIntel_README.md

```diff
@@ -0,0 +1,142 @@
+# ThreatIntel Module
+Python3 module that can be used in the development of Threat Intelligence Connectors for the Carbon Black Cloud.
+
+## Requirements/Installation
+
+The file `requirements.txt` contains a list of dependencies for this project. After cloning this repository, run the following command from the `examples/threathunter/threat_intelligence` directory:
+
+```
+pip install -r ./requirements.txt
+```
+
+
+## Introduction
+This document describes how to use the ThreatIntel Python3 module for development of connectors that retrieve Threat Intelligence and import it into a Carbon Black Cloud instance. This document describes how to use the ThreatIntel Python3 module for development of connectors that retrieve Threat Intelligence and import it into a Carbon Black Cloud instance. Documentation on Feed and Report definitions is [available here.](https://developer.carbonblack.com/reference/carbon-black-cloud/cb-threathunter/latest/feed-api/#definitions)
+
+The ThreatIntel `push_to_cb` method expects a list of objects that have six attributes:
+`id, timestamp, title, description, severity, and iocs_v2`. For these objects, may use the pre-built `AnalysisResult` class from `results.py`, which has these attributes already, or build your own custom class.
+
+### Files
+
+#### `threatintel.py`
+
+Contains the `ThreatIntel` class. This converts Threat Intelligence from result dictionaries into ThreatHunter `Reports`, and then sends them to a ThreatHunter `Feed`.
+
+#### `results.py`
+
+Contains the `AnalysisResult` and `IOC_v2` classes. This contains the logic to attach `iocs_v2` dictionaries to `AnalysisResult` objects.
+
+### Important ThreatIntel Methods
+
+#### `ThreatIntel.push_to_cb(feed_id, results)`:
+Input a ThreatHunter Feed ID and a list of result objects. The result objects must contain the following required attributes: `id, timestamp, title, description, severity, and iocs_v2`. Formats the result objects into ThreatHunter `Report` objects, calls the input validation function, then sends the `Report` objects to a ThreatHunter `Feed`.
+
+#### `ThreatIntel.input_validation(reports)`:
+Input a list of `Report` dictionaries, and outputs a boolean value indicating if the dictionaries are properly formatted for submission to a ThreatHunter `Feed`.
+
+
+## Data Classes
+
+### ThreatHunter Report
+CBAPI includes the `cbapi.psc.threathunter.Report` class. ThreatHunter `Report` objects have the following required and optional parameters:
+
+|Required|Type|Optional|Type|
+|---|---|---|---|
+|`id`|string|`link`|string|
+|`timestamp`|integer|`[tags]`|[str]|
```

Comment:
I don't believe timestamp is required as that should be generated when the report is submit.

Reply:
The ThreatHunter Report format (https://developer.carbonblack.com/reference/carbon-black-cloud/cb-threathunter/latest/feed-api/#definitions) lists timestamp as a required parameter. However, STIX data has embedded timestamps. Would it make more sense to timestamp the data as when it is sent to CB?

In the STIX/TAXII example, we're using the embedded timestamps, which represent the definition time for a specific indicator

Reply:
We've settled on the timestamp from the TAXII server